### PR TITLE
Convert unauthorized filesTest to pact.io

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
- Converted unauthorized filesTest to Pact.io

## Related Issue
- Part of https://github.com/owncloud/owncloud-sdk/issues/500

### CODE REVIEW

File: tests/pactHelper.js [hunk 1]
```diff
@@ -79,7 +79,7 @@ function setGeneralInteractions (provider) {
       }),
       headers: {
         'Access-Control-Request-Method': Pact.Matchers.regex({
-          matcher: 'GET|POST|PUT|DELETE|MKCOL|PROPFIND',
+          matcher: 'GET|POST|PUT|DELETE|MKCOL|PROPFIND|MOVE|COPY',
```
Comment (swoichha) [inline tests/pactHelper.js:82]: IMO we don't need `MOVE|COPY` for now. We can add them later when we add interactions for them.

Comment (kiranparajuli589) [reply tests/pactHelper.js:82]: interactions are added for move and copy request too.,

**Review by swoichha — COMMENTED** [review]
LGTM

File: tests/unauthorized/filesTest.js [hunk 2]
```diff
@@ -1,34 +1,129 @@
...
+  const expectedUnAuthorizedMessage = 'Username or password was incorrect, Username or password was incorrect'
+  const incorrectAuthorizationXmlResponseBody = '<?xml version="1.0" encoding="utf-8"?>\n' +
+    '<d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">\n' +
+    '  <s:exception>Sabre\\DAV\\Exception\\NotAuthenticated</s:exception>\n' +
+    '  <s:message>Username or password was incorrect, Username or password was incorrect</s:message>\n' +
+    '</d:error>'
+  const webdavFilesResponseHeader = {
+    'Access-Control-Allow-Origin': origin,
+    'Content-Type': 'application/xml; charset=utf-8',
+    'Access-Control-Allow-Methods': 'GET,OPTIONS,POST,PUT,DELETE,MKCOL,PROPFIND,PATCH,PROPPATCH,REPORT,HEAD,COPY,MOVE'
```
Comment (individual-it) [inline tests/unauthorized/filesTest.js:25]: why not using `accessControlAllowMethods`?

File: tests/unauthorized/filesTest.js [hunk 3]
```diff
@@ -1,34 +1,129 @@
...
   // TESTING CONFIGS
-  var testContent = 'testContent'
-  var testFolder = '/testFolder' + timeRightNow
-  var testSubFiles = [
-    testFolder + '/' + 'file one.txt',
-    testFolder + '/' + 'zz+z.txt',
-    testFolder + '/' + '中文.txt',
-    testFolder + '/' + 'abc.txt',
-    testFolder + '/' + 'subdir/in dir.txt'
+  const testContent = config.testContent
```
Comment (individual-it) [inline tests/unauthorized/filesTest.js:119]: why redefining the variables and not just use `config.` in all places?

File: tests/unauthorized/filesTest.js [hunk 4]
```diff
@@ -1,34 +1,129 @@
...
+    promises.push(provider.addInteraction({
+      uponReceiving: 'a request to copy a file with invalid authentication',
+      withRequest: {
+        method: 'COPY',
+        path: webdavUrl,
+        headers: requestHeaderWithInvalidAuth
+      },
+      willRespondWith: unauthorizedResponse
+    }))
+
```
Comment (individual-it) [inline tests/unauthorized/filesTest.js:110]: all those could be summarised in one single interaction using matchers https://github.com/pact-foundation/pact-js/#matching

**Review by individual-it — CHANGES_REQUESTED** [review]

File: tests/unauthorized/filesTest.js (hunk 2 above)
Comment (kiranparajuli589) [reply tests/unauthorized/filesTest.js]: done!

File: tests/unauthorized/filesTest.js (hunk 3 above)
Comment (kiranparajuli589) [reply tests/unauthorized/filesTest.js]: done

File: tests/unauthorized/filesTest.js (hunk 4 above)
Comment (kiranparajuli589) [reply tests/unauthorized/filesTest.js]: matchers seem to be not working for `method` so I've done refactoring with a for loop.

File: tests/unauthorized/filesTest.js [hunk 5]
```diff
@@ -36,77 +36,19 @@ fdescribe('Unauthorized: Currently testing files management,', function () {
...
+      promises.push(provider.addInteraction({
+        uponReceiving: `a ${method} request on file contents with invalid authentication`,
+        withRequest: {
+          method: method,
+          path: webdavUrl,
+          headers: requestHeaderWithInvalidAuth
+        },
+        willRespondWith: unauthorizedResponse
+      }))
+    })
```
Comment (individual-it) [inline tests/unauthorized/filesTest.js:51]:
```
provider.addInteraction({
      uponReceiving: 'a request to copy a file with invalid authentication',
      withRequest: {
        method: Pact.Matchers.term({
  matcher: 'GET|PUT|MKCOL|DELETE|COPY|MOVE|PROPFIND',
  generate: 'COPY'
}),
        path: webdavUrl,
        headers: requestHeaderWithInvalidAuth
      },
      willRespondWith: unauthorizedResponse
    })
```

Comment (individual-it) [reply tests/unauthorized/filesTest.js:51]: no loop needed

Comment (kiranparajuli589) [reply tests/unauthorized/filesTest.js:51]: I've also tried to use regex to match `method`. But Pact is not happy.
Screenshot from 2020-10-06 19-00-38

Comment (individual-it) [reply tests/unauthorized/filesTest.js:51]: sorry I haven't read your comment above

**Review by individual-it — APPROVED** [review]

**Pull request merged**